src/controllers/controller.line.js

```diff
@@ -148,46 +165,33 @@ module.exports = DatasetController.extend({
 	/**
 	 * @private
 	 */
-	_resolveElementOptions: function(point, index) {
+	_resolveElementOptions: function(element, elementName, optionNameMap, index, scriptable) {
 		var me = this;
 		var chart = me.chart;
 		var datasets = chart.data.datasets;
 		var dataset = datasets[me.index];
-		var custom = point.custom || {};
-		var options = chart.options.elements.point;
+		var custom = element.custom || {};
+		var elementOpts = chart.options.elements[elementName];
 		var values = {};
 		var i, ilen, key;
 
 		// Scriptable options
-		var context = {
+		var context = scriptable && {
 			chart: chart,
 			dataIndex: index,
 			dataset: dataset,
 			datasetIndex: me.index
 		};
 
-		var ELEMENT_OPTIONS = {
-			backgroundColor: 'pointBackgroundColor',
-			borderColor: 'pointBorderColor',
-			borderWidth: 'pointBorderWidth',
-			hitRadius: 'pointHitRadius',
-			hoverBackgroundColor: 'pointHoverBackgroundColor',
-			hoverBorderColor: 'pointHoverBorderColor',
-			hoverBorderWidth: 'pointHoverBorderWidth',
-			hoverRadius: 'pointHoverRadius',
-			pointStyle: 'pointStyle',
-			radius: 'pointRadius',
-			rotation: 'pointRotation',
-		};
-		var keys = Object.keys(ELEMENT_OPTIONS);
+		var keys = Object.keys(optionNameMap);
 
 		for (i = 0, ilen = keys.length; i < ilen; ++i) {
 			key = keys[i];
 			values[key] = resolve([
 				custom[key],
-				dataset[ELEMENT_OPTIONS[key]],
+				dataset[optionNameMap[key]],
 				dataset[key],
-				options[key]
+				elementOpts[key]
 			], context, index);
```

Comment:
We should avoid the extra `dataset[optionNameMap[key]]` which is always the same as `dataset[key]` for line (except for tension). Duplicating this method would be the easier way to do that (see previous comment).